src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -343,3 +346,165 @@ class Config:
+        else:
+            data["extra"] = ""
+
+        if data.get("month"):
+                if len(data['month']) == 12:
+                    data['month'] = '*'
+                else:
+                    data['month'] = ",".join(data['month'])
+
+        if data.get("dayweek"):
+                if len(data['dayweek']) == 7:
+                    data['dayweek'] = '*'
+                else:
+                    data['dayweek'] = ",".join(data['dayweek'])
+
+        mode = data.get("mode")
+        rmodule = data.get("remotemodule")
+        if mode == 'module' and not rmodule:
+            verrors.add('remotemodule', 'This field is required')
+
+        rpath = data.get("remotepath")
+        if mode == 'ssh' and not rpath:
+            verrors.add('remotepath', 'This field is required')
+
+        username = data.get("user")
+        if mode == 'ssh':
+            rport = data.get("remoteport")
+            if not rport:
+                verrors.add('remoteport', 'This field is required')
+  
+            try:
+                user = await self.middleware.call('user.query', [('username', '=', username)], {'get': True})
+            except IndexError:
+                verrors.add('user', 'Provided user does not exists')
+            else:
+                search = os.path.join(user['home'], ".ssh", "id_[edr]*.*")
+                if not glob.glob(search):
+                    verrors.add(
+                        "user",
+                        "In order to use rsync over SSH you need a user"
+                        "with a public key (DSA/ECDSA/RSA) set up in home dir."
+                    )
+                else:
+                    if data.get("validate_rpath"):
+                        rhost = str(data.get("remotehost"))
+                        if '@' in rhost:
+                            username, rhost = rhost.split('@')
+
+                        try:
+                            async with asyncssh.connect(
+                                rhost,
+                                port=rport,
+                                username=username) as conn:
+
+                                await conn.run(f'test -d {rpath}', check=True)
+
+                        except (OSError, asyncssh.Error) as e:
+                            verrors.add(
+                                "remotepath",
+                                "The Remote Path you specified does not exist or is not a "
+                                "directory.Either create one yourself on the remote "
+                                "machine or uncheck the rsync_validate_rpath' field."
+                                "**Note**: This could also happen if the remote path "
+                                "entered exceeded 255 characters and was truncated, please"
+                                " restrict it to 255. Or it could also be that your SSH "
+                                "credentials (remote host,etc) are wrong."
```

Comment:
It would be great to somehow inspect `asyncssh.Error` and tell user what is exactly wrong.  At least provide `ssh` command stderr. I've tried to use this while investigating https://redmine.ixsystems.com/issues/27950 and had a very bad time.